src/main/java/org/folio/helper/CheckinReceivePiecesHelper.java

```diff
@@ -472,9 +249,60 @@ private Future<Void> storeUpdatedPieceRecord(Piece piece, RequestContext request
       });
   }
 
-
-  protected List<String> getPoLineIdsForUpdatedPieces(Map<String, List<Piece>> piecesGroupedByPoLine) {
-    return EntryStream
+  /**
+   * @param pieceId piece id
+   * @return PO Line id corresponding to passed pieceId from request data
+   */
```

Comment:
these comments is not useful from my prespective

Reply:
Change from first commit - I understand many things can be simplified but we can't do all this in scope of one commit